Add a command to allow querying whether the VM is migratable

Add a command to allow querying whether the VM is migratable which
allows the toolstack to prevent migrations when it won't work.
Like many QMP commands, it returns nothing on success (if the VM is
migratable). If the VM is not migratable, it returns an error message
indicating which device is preventing migration.
diff --git a/migration/savevm.c b/migration/savevm.c
index 3b865752d0..3b8a6a918e 100644
--- a/migration/savevm.c
+++ b/migration/savevm.c
@@ -2207,6 +2207,11 @@ int save_snapshot(const char *name, Error **errp)
     return ret;
 }
 
+void qmp_query_migratable(Error **errp)
+{
+    qemu_savevm_state_blocked(errp);
+}
+
 void qmp_xen_save_devices_state(const char *filename, Error **errp)
 {
     QEMUFile *f;
diff --git a/qapi-schema.json b/qapi-schema.json
index 417a8f1928..c0e4afbd3e 100644
--- a/qapi-schema.json
+++ b/qapi-schema.json
@@ -3261,6 +3261,23 @@
 ##
 { 'command': 'migrate-incoming', 'data': {'uri': 'str' } }
 
+##
+# @query-migratable:
+#
+# Query whether a VM is migratable or not.
+#
+# Returns: Nothing on success
+#
+# Since: XXX.XXX
+#
+# Example:
+#
+# -> { "execute": "query-migratable" }
+# <- { "return": { } }
+#
+##
+{ 'command': 'query-migratable' }
+
 ##
 # @xen-save-devices-state:
 #
